lms/djangoapps/mobile_api/video_outlines/serializers.py

```diff
@@ -159,23 +159,33 @@ def find_urls(course_id, block, child_to_parent, request):
     return unit_url, section_url
 
 
-def video_summary(course, course_id, video_descriptor, request, local_cache):
+def video_summary(video_profiles, course_id, video_descriptor, request, local_cache):
     """
     returns summary dict for the given video module
     """
-    # First try to check VAL for the URLs we want.
-    val_video_info = local_cache['course_videos'].get(video_descriptor.edx_video_id, {})
-    if val_video_info:
-        video_url = val_video_info['url']
+    # Get encoded videos
+    encoded_videos = local_cache['course_videos'].get(video_descriptor.edx_video_id, {})
+
+    # Get highest priority video to populate backwards compatible field
+    default_encoded_video = {}
+
+    if encoded_videos:
+        for profile in video_profiles:
+            if encoded_videos['profiles'].get(profile):
+                default_encoded_video = encoded_videos['profiles'].get(profile)
+                break
+
+    if default_encoded_video:
+        video_url = default_encoded_video['url']
```

Comment:
Need to change this to `default_encoded_video.get('url'`).
The reason being that we never know whether VEDA has completed the video processing of all encodings of all the videos in a course.  We are usually very defensive in our calling of VAL code.

Once this change is made, the lines below should first check whether `video_url` has been set.